### PR TITLE
#162633482 Like-Unlike a comment

### PR DESCRIPTION
**Pull request by mycok**
**What does this PR do?**

Adds functionality to like or unlike a comment

**Description of Task to be completed?**

- Add like button to the interface
- Add toggle functionality to change button colour whenever a user likes or unlikes
- Increment comment likes and dislikes dynamically

**How should this be manually tested?**

- Run `npm install` to install dependencies.
- Run `npm start` to run the application.

**Any background context you want to provide?**

- In case there are no articles, create an article and a comment in order to test the functionality

**What are the relevant pivotal tracker stories?**

[#162633482](https://www.pivotaltracker.com/story/show/162633482)

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,4 +1,4 @@
 PORT=8000
-HOST="https://ah-backend-valkyrie-staging.herokuapp.com/api/v1"
+HOST="http://ah-backend-valkyrie-staging.herokuapp.com/api/v1"
```
Comment (mubstimor) [inline .env]: why change the host back to http when the group agreed we should use https.

File: src/components/comments/actions/AsyncActions.test.js [hunk 2]
```diff
@@ -1,107 +1,106 @@
...
+      req.respondWith({
+        status: 204,
+        response: comment,
+      });
+    });
+    const message = "comment 2 has been successfully deleted";
+    const expectedAction = [{
+      type: commentsActionTypes.DELETE_COMMENT,
+      message,
+    }];
```
Comment (mubstimor) [inline src/components/comments/actions/AsyncActions.test.js:101]: blank line looks out of place here

File: src/reducers/rootReducer.js [hunk 3]
```diff
@@ -16,5 +16,5 @@ export default combineReducers({
   articles: articleReducer,
   errors: errorReducer,
   comments: commentReducer,
-  rated: rateReducer,
```
Comment (mubstimor) [inline src/reducers/rootReducer.js]: please add back the rated reducer, Its part of Baker's functionality

**Review by mubstimor — CHANGES_REQUESTED** [review]
please revise the flagged changes.

**Review by mubstimor — APPROVED** [review]

File: src/components/comments/actions/action_types.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
+  FETCH_COMMENTS_ERROR: "FETCH_COMMENTS_ERROR",
+  ADD_COMMENT: "ADD_COMMENT",
+  LIKE_COMMENT: "LIKE_COMMENT",
+  LIKE_COMMENT_ERROR: "LIKE_COMMENT_ERROR",
+  ADD_COMMENT_ERROR: "ADD_COMMENT_ERROR",
+  DELETE_COMMENT: "DELETE COMMENT",
+  DELETE_COMMENT_ERROR: "DELETE_COMMENT_ERROR",
+  EDIT_COMMENT_SUCCESS: "EDIT_COMMENT_SUCCESS",
+  EDIT_COMMENT_ERROR: "EDIT_COMMENT_ERROR",
+
```
Comment (NabunyaLilian) [inline src/components/comments/actions/action_types.js:12]: Please remove this line

**Comment by NabunyaLilian** [comment]
@mycok , please work on your description of tasks section in the PR by providing the tasks you did.

**Comment by NabunyaLilian** [comment]
Otherwise good work, this LGTM.

File: src/assets/css/style.css [hunk 5]
```diff
@@ -83,11 +75,6 @@ img {
...
-.rate-star {
-  font-size: 18px;
-  color: gold;
+.cursor-pointer {
+  cursor: pointer;
 }
-
-.list-group a:hover {
-  background-color: #f9f9f9 !important;
-}
```
Comment (fatukunda) [inline src/assets/css/style.css:93]: Leave a line here.

File: src/components/comments/actions/actions.js [hunk 6]
```diff
@@ -58,44 +58,63 @@ export const editCommentFailure = error => (
...
 
-export const editComment = ( slug, id, commentToEdit ) => ( dispatch ) => {
-   return api.updateResource( `${ commentsUrl }${ slug }/comments/${ id }`, commentToEdit )
-    .then( ( response ) => {
-      dispatch( editCommentSuccess( { response, commentToEdit, id } ) );
-    } )
-    .catch( ( err ) => {
-      dispatch( editCommentFailure( err.response.data.errors.body ) );
-    } );
-};
```
Comment (fatukunda) [inline src/components/comments/actions/actions.js:101]: A line is needed here

**Comment by ruganda** [comment]
@mycok , Please resolve the conflicts

**Comment by fatukunda** [comment]
LGTM

**Pull request merged**